pkg/metrics/metrics_test.go

```diff
@@ -0,0 +1,264 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"math"
+	"net/http"
+	"os"
+	"reflect"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	cmg "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/expfmt"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var (
+	mgr MetricsManager
+	srv *http.Server
+	wg  *sync.WaitGroup
+)
+
+const (
+	httpPattern = "/metrics"
+	addr        = "localhost:12134"
+	httpAddr    = "http://" + addr + httpPattern
+)
+
+func initMgr() {
+	wg = &sync.WaitGroup{}
+	wg.Add(1)
+	mgr = NewMetricsManager()
+	var err error
+	if srv, err = mgr.StartServing(httpPattern, addr, nil, wg); err != nil {
+		log.Fatalf("failed to start serving [%v]", err)
+	}
+}
+
+func shutdown() {
+	if err := srv.Shutdown(context.Background()); err != nil {
+		panic(err)
+	}
+	wg.Wait()
+}
+
+func TestMain(m *testing.M) {
+	initMgr()
+	c := m.Run()
+	shutdown()
+	os.Exit(c)
+}
+
+func TestNew(t *testing.T) {
+	if mgr == nil {
+		t.Errorf("failed testing new")
+	}
+}
+
+func TestStartServing(t *testing.T) {
+	if _, err := http.Get(httpAddr); err != nil {
+		t.Errorf("failed to get response from server %v", err)
+	}
+}
+
+func TestDropOperation(t *testing.T) {
```

Comment:
Can we also have some test cases like multiple operations at the same time for the same driver, in various states, some dropped, some failed, some success, some ongoing?